sources/Renderer/Metal/RenderState/MTGraphicsPSO.mm

```diff
@@ -192,14 +193,17 @@ static void FillColorAttachmentDesc(
     if (hasTessellationStage)
     {
         vertexShaderMT = LLGL_CAST(const MTShader*, desc.tessEvaluationShader);
-        if (vertexShaderMT == nullptr || !vertexShaderMT->IsPostTessellationVertex())
-            throw std::invalid_argument("cannot create Metal tessellation pipeline without post-tessellation vertex shader (in 'tessEvaluationShader')");
+
+        LLGL_THROW_IF(
+            vertexShaderMT == nullptr || !vertexShaderMT->IsPostTessellationVertex(),
+            std::invalid_argument("cannot create Metal tessellation pipeline without post-tessellation vertex shader (in 'tessEvaluationShader')")
+        );
     }
     else
     {
         vertexShaderMT = LLGL_CAST(const MTShader*, desc.vertexShader);
-        if (vertexShaderMT == nullptr)
-            throw std::invalid_argument("cannot create Metal pipeline without vertex shader");
+
+        LLGL_THROW_IF(vertexShaderMT == nullptr, std::invalid_argument("cannot create Metal pipeline without vertex shader"));
```

Comment:
There are still several uses of this new macro in the `sources/` folder. I'm okay with replacing the `std::invalid_argument` exception with `LLGL_TRAP()` or `LLGL_ASSERT()`. My rationale here is that LLGL is supposed to use very little exceptions anyway and the little number of exceptions that are thrown can be unified with a thin abstraction to provide helpful callstacks when exceptions are disabled.